File: tasking/sources/taskloop_dep.2.f90
! @@name:	taskloop_dep.2
! @@type:	F-free
! @@operation:	compile
! @@expect:	success
! @@version:	omp_6.0
subroutine process_work_b(n, B)
   implicit none
   integer :: n
   real :: B(n,*)
   integer :: i, j

   ! Dependences for taskloop iterations in multi-dimensional loop nest

   ! TL4 taskloop + collapse
   !$omp taskloop collapse(2) nogroup
   do j = 2, n
      do i = 2, n
         !$omp task_iteration depend(inout: B(i,j))  &
         !$omp&               depend(in: B(i-1,j), B(i,j-1))
         B(i,j) = B(i,j) + B(i,j) * B(i-1,j) * B(i,j-1)
      end do
   end do
   !$omp end taskloop

   ! T5 other task
   !$omp task depend(in: B(n,n))
   print *, "B(n,n) =", B(n,n)
   !$omp end task
end subroutine

File: tasking/sources/taskloop_dep.2.c
/*
* @@name:	taskloop_dep.2
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
#include <stdio.h>

void process_work_b(int n, float *B[n])
{
   // Dependences for taskloop iterations in multi-dimensional loop nest

   // TL4 taskloop + collapse
   #pragma omp taskloop collapse(2) nogroup
   for (int i = 1; i < n; i++)
   {
      for (int j = 1; j < n; j++)
      {
         #pragma omp task_iteration depend(inout: B[i][j]) \
                                    depend(in: B[i-1][j], B[i][j-1])
         B[i][j] += B[i][j] * B[i-1][j] * B[i][j-1];
      }
   }

   // T5 other task
   #pragma omp task depend(in: B[n-1][n-1])
   printf("B[n-1][n-1] = %f\n", B[n-1][n-1]);
}
